app/assets/javascripts/Components/Modals/create_modify_annotation_panel_modal.jsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import Modal from "react-modal";
+
+class CreateModifyAnnotationPanel extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      content: this.props.content,
+      category_id: this.props.category_id,
+    };
+  }
+
+  componentDidMount() {
+    Modal.setAppElement("body");
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps !== this.props) {
```

Comment:
Hmm, in general I prefer just comparing specific prop values (in this case, `content` and `category_id`.

Slightly related: one request we had was to preserve form data if the TA accidentally closed the modal (by pressing Esc or just clicking Cancel instead of Save), so that the next time the modal is opened in those cases, the previous annotation text is still there. I think if you make the change I described here, that should happen---but please test.

Reply:
Yep, you were right. Though it took me a decent while to identify why this was the case. Really impressed that you identified this at a glance!